pkg/providers/instancetype/types.go

```diff
@@ -192,6 +233,23 @@ func ephemeralStorage(amiFamily amifamily.AMIFamily, blockDeviceMappings []*v1al
 	return amifamily.DefaultEBS.VolumeSize
 }
 
+func privateIPv4Address(name string) *resource.Quantity {
+	// https://docs.aws.amazon.com/eks/latest/userguide/windows-support.html
+	// The number of pods that you can run per Windows node
+	// is equal to the number of IP addresses available per elastic network interface for the node's instance type,
+	// minus one.
+	// however, https://aws.github.io/aws-eks-best-practices/windows/docs/networking/
```

Comment:
For Windows, IPs are managed by VPC Resource controller and therefore, the limit is `Number of supported IPs - 1`. 
Reference: https://github.com/aws/amazon-vpc-resource-controller-k8s/blob/83e4e5f88ede891e05b3d7f80a9cc0633df07567/pkg/provider/ip/provider.go#L290

Also, why do we need this value?

As new features are launched for Windows, such as multi-eni support or Prefix delegation support, this value would change based on a lot of factors and would be partially customer configurable. Such information might not be present with Karpenter. Therefore, maybe we should work around this.

Reply:
1. This is a must for Windows pod, because "vpc.amazonaws.com/PrivateIPv4Address" is a resource in pod.spec.resource.requirement. And Karpenter must define such capability for Windows pod, otherwise the pod's requirement is not met.
2. From Karpenter perspective, the capacity must minus 3, since the IP addresses are consumed by the node, CNI and kube-proxy already.

Reply:
I'm also not clear on this. We are aware of the DS that exist in your cluster. It's possible that a user isn't running `vpc-cni` or isn't running `kube-proxy`, in which case, the `- 3` calculation would be off. This should be `- 1` as @rawahars said and Karpenter will handle the pod calculation for the DS.

Reply:
> * This is correct that `vpc.amazonaws.com/PrivateIPv4Address` is a required resource. How is the value calculated here used by Karpenter?
> * For Windows nodes, the IPs are not consumed by CNI and kube-proxy since they run on the node itself. I'm not sure what you mean by Karpenter perspective.


Below is a node with only one Windows pod. 3 secondary private IPv4 addresses are allocated here. One is for the pod. I guess CNI and kube-proxy consume another two.

<img width="296" alt="image" src="https://user-images.githubusercontent.com/1520380/230322611-963c57bf-d6f2-4ea0-9352-3929e99894fd.png">

Reply:
ah no that's not the case. Basically, at any time, VPC Resource Controller would keep 3 or so IPs warmed up on the node so that as soon as a pod is created, an IP can be allocated to it. The additional secondary IPs which you see are the IPs which have been warmed up to be allocated to the pods.

There would be one primary IP for the interface which is not used by the pods (Hence -1 in the calculation).